next-docs/pages/store/tables.mdx

```diff
@@ -1 +1,149 @@
+import { Callout } from "nextra/components";
+import TableIntro from "./table-intro.mdx";
+
 # Tables
+
+<TableIntro />
+
+Tables are registered in the `Store` contract at runtime.
+When a table is registered, its ID, key and value types, as well as key and field names are stored in the internal `Tables` table.
+This emits an event that can be used by [offchain indexers](../services/indexer) to start replicating the state of the new table.
+
+The recommended way of reading from tables and writing to tables is via the [typed table libraries](./table-libraries).
+However, it is also possible to use the low-level `IStore` (external) or `StoreCore` (internal) [API](./reference) directly.
+
+## Types of tables
+
+There are two types of tables in `Store`: _Onchain tables_ and _offchain tables_.
+We often omit the prefix from onchain tables and just call them tables.
+
+As the name suggests, **onchain tables** store their state onchain, in the `Store` contract.
+In addition, an event is emitted on every write operation, to allow [offchain indexers](../services/indexer) to replicate the onchain state.
+
+**Offchain tables** on the other hand don't store any state onchain,
+but only emit the events for [offchain indexers](../services/indexer).
+This makes them suitable for use cases where data doesn't need to be retrieved onchain,
+but should still be synchronized from the `Store` contract to [offchain indexers](../services/indexer).
+
+Onchain you can write to offchain tables with the same methods as onchain tables,
+except for reading data and modifying partial data in a dynamic field.
+
+## Advanced
+
+<Callout type="info" emoji="ⓘ">
+  In practice, everything in this section is automatically handled by the [typed table libraries](./table-libraries).
+  Read on if you're curious about the inner workings.
+</Callout>
+
+### Table ID
+
+The table ID encodes the table's type and name in a 32-byte `ResourceId` value and uniquely identifies a table.
+
+```solidity copy
+import { ResourceId, ResourceIdLib } from "@latticexyz/store/src/ResourceId.sol";
+import { RESOURCE_TABLE, RESOURCE_OFFCHAIN_TABLE } from "@latticexyz/store/src/storeResourceTypes.sol";
+
+// Using the `RESOURCE_TABLE` type makes it an onchain table
+ResourceId tableId = ResourceIdLib.encode({
+  typeId: RESOURCE_TABLE,
+  name: "Balance"
+});
+
+// Using the `RESOURCE_OFFCHAIN_TABLE` type makes it an offchain table
+ResourceId offchainTableId = ResourceIdLib.encode({
+  typeId: RESOURCE_OFFCHAIN_TABLE,
+  name: "Balance"
+});
+```
+
+### Key schema
+
+The key schema represents the data types of the (composite) key of the table.
+Internally the composite key is always handled as `bytes32[]`,
+but the key schema allows [offchain indexers](../services/indexer) to create properly typed composite key columns,
+and allows MUD to generate [typed table libraries](./table-libraries).
+
+The key schema can contain up to 28 static data types.
```

Comment:
We use "static" and "dynamic" in an unusual sense, so the first time we use either on a page I think it's best to explain what we mean.